maas-schemas/schemas/core/components/fare.json

```diff
@@ -1,38 +1,172 @@
+      "type": "object",
+      "properties": {
+        "currency": { "$ref": "http://maasglobal.com/core/components/units.json#/definitions/metaCurrencyTOKEN" },
+        "tokenId": {
+          "$ref": "#/definitions/tokenId"
+        },
+        "amount": { "$ref": "#/definitions/tokenAmount" },
+        "originalAmount": {
+          "$ref": "#/definitions/tokenAmount"
+        },
+        "productionAmount": {
+          "$ref": "#/definitions/tokenAmount"
+        }
+      },
+      "required": ["tokenId"]
+    },
+    "abstractBalance": {
+      "type": "object",
+      "properties": {
+        "currency": { "$comment": "placeholder" },
+        "tokenId": { "$comment": "placeholder" },
+        "amount": { "$comment": "placeholder" }
+      },
+      "required": ["currency", "amount"],
+      "additionalProperties": false
+    },
+    "wmpBalance": {
+      "description": "An amount of in-app credits with type information",
+      "allOf": [{ "$ref": "#/definitions/abstractBalance" }, { "$ref": "#/definitions/wmpFields" }],
+      "examples": [
+        {
+          "currency": "WMP",
+          "amount": 1234
+        }
+      ]
+    },
+    "tokenBalance": {
+      "description": "An amount of in-app tokens with type information",
+      "allOf": [{ "$ref": "#/definitions/abstractBalance" }, { "$ref": "#/definitions/tokenFields" }],
+      "examples": [
+        {
+          "currency": "TOKEN",
+          "tokenId": "cx-test-token_v2",
+          "amount": 1
+        }
+      ]
+    },
+    "balance": {
+      "oneOf": [{ "$ref": "#/definitions/wmpBalance" }, { "$ref": "#/definitions/tokenBalance" }]
+    },
+    "balancesArray": {
+      "type": "array",
+      "items": {
+        "$ref": "#/definitions/balance"
+      }
+    },
+    "balancesObject": {
+      "description": "A summary of in-app credit and tokens",
+      "type": "object",
+      "properties": {
+        "WMP": { "$ref": "#/definitions/wmpBalance" }
+      },
+      "patternProperties": {
+        "^[a-z]+(-[a-z]+)*-[a-z0-9_]+$": { "$ref": "#/definitions/tokenBalance" }
+      },
+      "required": ["WMP"],
+      "additionalProperties": false,
+      "examples": [
+        {
+          "WMP": {
+            "currency": "WMP",
+            "amount": 1234,
+            "type": "charge"
+          },
+          "cx-test-token_v2": {
+            "currency": "TOKEN",
+            "tokenId": "cx-test-token_v2",
+            "amount": 1,
+            "type": "charge"
+          }
+        }
+      ]
+    },
+    "abstractFare": {
+      "type": "object",
+      "properties": {
+        "type": { "$ref": "#/definitions/fareType" },
+        "currency": { "$comment": "placeholder" },
+        "tokenId": { "$comment": "placeholder" },
+        "amount": { "$comment": "placeholder" },
+        "originalAmount": { "$comment": "placeholder" },
+        "productionAmount": { "$comment": "placeholder" }
+      },
+      "required": ["type", "currency", "amount"],
```

Comment:
Maybe this is too loose? Maybe we should make more fields compulsory since balance and fare are now separate.